integration-libs/cdc/root/service/cdc-js.service.spec.ts

```diff
@@ -572,6 +582,233 @@ describe('CdcJsService', () => {
+        'setAccountInfo'
+      ).and.callFake((options: { callback: Function }) => {
+        options.callback({ status: 'OK' });
+      });
+      expect(service.updateUserPasswordWithoutScreenSet).toBeTruthy();
+      let oldPass = 'OldPass123!';
+      let newPass = 'Password1!';
+
+      service
+        .updateUserPasswordWithoutScreenSet(oldPass, newPass)
+        .subscribe(() => {
+          expect(
+            winRef.nativeWindow['gigya'].accounts.setAccountInfo
+          ).toHaveBeenCalledWith({
+            password: oldPass,
+            newPassword: newPass,
+            callback: jasmine.any(Function),
+          });
+        });
+      done();
+    });
+  });
+
+  describe('updateUserEmailWithoutScreenSet', () => {
+    it('should not call updateUserEmailWithoutScreenSet', (done) => {
+      spyOn(
+        winRef.nativeWindow['gigya'].accounts,
+        'setAccountInfo'
+      ).and.callFake((options: { callback: Function }) => {
+        options.callback({ status: 'OK' });
+      });
+      expect(service.updateProfileWithoutScreenSet).toBeTruthy();
+      service.updateUserEmailWithoutScreenSet('', '').subscribe(() => {
+        expect(
+          winRef.nativeWindow['gigya'].accounts.setAccountInfo
+        ).not.toHaveBeenCalled();
+      });
+      done();
+    });
+
+    it('should call updateUserEmailWithoutScreenSet', (done) => {
+      spyOn(
+        winRef.nativeWindow['gigya'].accounts,
+        'setAccountInfo'
+      ).and.callFake((options: { callback: Function }) => {
+        options.callback({ status: 'OK' });
+      });
+      spyOn(winRef.nativeWindow['gigya'].accounts, 'login').and.callFake(
+        (options: { callback: Function }) => {
+          options.callback({ status: 'OK' });
+        }
+      );
+
+      expect(service.updateUserEmailWithoutScreenSet).toBeTruthy();
+      let pass = 'Password123!';
+
+      service.updateUserEmailWithoutScreenSet(pass, newEmail).subscribe(() => {
+        expect(
+          winRef.nativeWindow['gigya'].accounts.setAccountInfo
+        ).toHaveBeenCalledWith({
+          profile: {
+            email: newEmail,
+          },
+          callback: jasmine.any(Function),
+        });
+        expect(userProfileFacade.update).toHaveBeenCalledWith({
+          uid: newEmail,
+        });
+      });
+      done();
+    });
+  });
+
+  describe('getLoggedInUserEmail', () => {
+    it('should return the logged in user email', () => {
+      userProfileFacade.get = createSpy().and.returnValue(
+        of({ uid: newEmail })
+      );
+
+      expect(service['getLoggedInUserEmail']()).toEqual(newEmail);
+    });
+
+    it('should return empty if no email is obtained', () => {
+      userProfileFacade.get = createSpy().and.returnValue(of({}));
+
+      expect(service['getLoggedInUserEmail']()).toEqual('');
+    });
+  });
+
+  describe('updateAddressWithoutScreenSet', () => {
+    it('should not call updateAddressWithoutScreenSet', (done) => {
```

Comment:
Similar change suggestion applies to other places in this file:

```suggestion
    it('should not call accounts.setAccountInfo', (done) => {
```

Reply:
I think this would be confusing, since `accounts.setAccountinfo` is called for most updates in the User profile field, So it will be difficult to identify which test failed in the test suite on failure

Reply:
Although the nested `it()` would have similar description, the parent `describe()` would have different, so the combined description in case of failed report should be distinguishable, e.g.: 
`updateAddressWithoutScreenSet -> should not call accounts.setAccountInfo` 
? So we know it's related to `updateAddressWithoutScreenSet` method under test.

Maybe I'm missing something... but current test description it not too informative to me, e.g.:
`updateAddressWithoutScreenSet ->  should not call updateAddressWithoutScreenSet`. It's not even true: we acutally call this method, because it's a method under test.

Reply:
noted @Platonn, will make the suggested changes.